src/wp-admin/includes/user.php

```diff
@@ -445,6 +445,10 @@ function wp_delete_user( $id, $reassign = null ) {
 
 	clean_user_cache( $user );
 
+	foreach ( get_post_types() as $post_type ) {
+		clear_user_posts_count_cache( $user->ID, $post_type );
+	}
+
```

Comment:
This should also be added to `wpmu_delete_user()`, likely inside of the `get_blogs_for_user()` foreach loop because the blog & user posts are already being manipulated there.

https://github.com/WordPress/wordpress-develop/blob/7c6678f2c20a09edc10d10ab14665d26ebb01754/src/wp-admin/includes/ms.php#L176-L199

Note that this happens before `clean_user_cache()`, and I think that might be the better placement for this, too – it would avoid any _potential_ edge-case race-conditions with the `$user` object no longer having a cached value during the `clear_user_posts_count_cache()` calls, because functions like `wp_cache_delete()` are considered pluggable (hence predictably unpredictable).